src/main/java/org/spongepowered/api/util/command/source/ProxySource.java

```diff
@@ -24,20 +24,25 @@
  */
 package org.spongepowered.api.util.command.source;
 
-import com.google.common.base.Optional;
-import org.spongepowered.api.entity.Entity;
 import org.spongepowered.api.util.command.CommandSource;
 
 /**
  * Proxy sources are {@link CommandSource}s that are run by one source as a
- * different source.
+ * different object.
  */
-public interface ProxySource extends LocatedSource {
+public interface ProxySource extends CommandSource {
 
     /**
-     * Get the {@link CommandSource} this source is proxying. This
+     * Get the {@link CommandSource} this source is created by.
      *
      * @return The proxied source
      */
     CommandSource getOriginalSource();
+
+    /**
+     * Get the target of this source.
+     *
+     * @return The target
+     */
+    ProxySourceTarget getTarget();
```

Comment:
Actually, it's not. If PlayerA types `/execute PlayerB ~ ~ ~ /command`, `getOriginalSource()` returns PlayerA, while `getTarget()` returns PlayerB.